packages/lit-html/src/private-ssr-support.ts

```diff
@@ -57,6 +57,22 @@ export const _$LH = {
         return resolveOverrideFn(this, values);
       }
     },
+  patchDirectiveResolve: (
+    directiveClass: typeof Directive,
+    resolveOverrideFn: (
+      this: Directive,
+      _part: Part,
+      values: unknown[]
+    ) => unknown
+  ) => {
+    // TODO: This could be optimized such that `_$resolve` is only patched on
+    // the Directive base class once. This should reduce class shape changes and
+    // be slightly more optimal. The hazard is that we need to handle both the
+    // unminified and minified property keys.
```

Comment:
The minification comment seems either to also apply to this approach, and `overrideDirectiveResolve`.  If this `patchDirectiveResolve` function is called against a minified lit-html Directive class, it won't work, right? We should try to trigger that in a test.

Reply:
Andrew and I were discussing this in person yesterday. I believe the `overrideDirectiveResolve` works fine with minification specifically because it's the override method name is minified by the same process. Climbing the prototype and looking for that method could be difficult because `hasOwnProperty()` or `Object.hasOwn()` takes a string.

Reply:
The minification comment issue that I was expecting is specifically for string literal handling, e.g.: `hasOwnProperty("_$resolve")`. Where I'm not sure `_$resolve` will become `S`. But should check.

Reply:
There's a separate minification issue where the version of lit-html that ssr imports could be minified and the version that contains the directive not minified, or vice-versa. This could happen with say, older versions of Shoelace that bundle the production build, or any situation where the client-side app is built separately from the server app.

Reply:
Ah yeah. I think I have actually encountered that issue and it currently exists.